### DISCUSSION TITLE
Why change `'formatoptions'` in filetype plugins?

### DISCUSSION THREAD

**Discussion by Danielkonge** [neovim/neovim#26908] [General]
This is related to a discussion question I asked yesterday: https://github.com/neovim/neovim/discussions/26885

Why is neovim changing the formatoptions in many of the `ftplugin` files? I could maybe understand removing known problematic options in certain languages, but the files often also add multiple options. 

Is this something carried over from Vim itself? Or is this a choice that has been made by neovim maintainers? If so, what is the advantage of having these formatoptions changed by filetype plugins?

**Comment by clason** [comment]
These files are indeed taken verbatim from Vim. Clearly the individual maintainers felt that these options were appropriate and useful for the corresponding language. If you disagreed, feel free to contact the maintainers listed at the top of each file and state your case.

**Reply by Danielkonge** [reply]
I feel like this is something that there should be some consensus on for vim (and/or neovim), but if this is just taken from vim directly that consensus should probably be reached among vim maintainers first. (I guess the consensus for now is just that it is up to the individual maintainers to decide for each language, but I don't really see the benefit.) 

I will try to see if this was ever discussed in the vim community and possibly start a discussion or issue there.

**Reply by Danielkonge** [reply]
Update: Knowing that I should look for vim discussions instead of neovim discussion, it was not hard to find earlier discussions about this. E.g. https://github.com/vim/vim/issues/4680 and multiple stack overflow discussions. Also `:h ftplugin-overrule` describes a way to get around this, but I think just an autocmd for `Filetype *` is easier. 

It seems that the vim consensus is that `ftplugin` files should be able to change `formatoptions`, which I don't really agree with, but it doesn't seem likely that I will change any opinions on this...

**Comment by dundargoc** [comment]
Tbh I agree; filetype plugin files has no business changing formatoptions most of the time and is a terrible design choice IMO.
